bot/cogs/filtering.py

```diff
@@ -369,6 +398,48 @@ def mod_log(self) -> ModLog:
         except discord.errors.Forbidden:
             await channel.send(f"{filtered_member.mention} {reason}")
 
+    async def _scheduled_task(self, msg: dict) -> None:
+        """Delete an offensive message once its deletion date is reached."""
+        delete_at = dateutil.parser.isoparse(msg['delete_date'])
```

Comment:
If `msg['delete_date']` contains a datetime string with a timezone (like the ones returned by the API), then we should make sure to strip the tzinfo here like we do in L418.

Currently, we're getting this each time we try to delete a rescheduled message:

```py
Traceback (most recent call last):
  File "/home/sebastiaan/pydis/repositories/bot/bot/cogs/filtering.py", line 405, in _scheduled_task
    await wait_until(delete_at)
  File "/home/sebastiaan/pydis/repositories/bot/bot/utils/time.py", line 103, in wait_until
    delay = time - (start or datetime.datetime.utcnow())
TypeError: can't subtract offset-naive and offset-aware datetimes
```

This would solve that:

```suggestion
        delete_at = dateutil.parser.isoparse(msg['delete_date']).replace(tzinfo=None)
```